src/rpc.rs

```diff
@@ -14,21 +14,24 @@ pub struct Glob {
 impl Glob {
     pub fn match_shv_ri(&self, shv_ri: &ShvRI) -> bool {
         // if method is granted => signal is granted as well
-        if self.path.matches(shv_ri.path()) && self.method.matches(shv_ri.method()) {
-            // path and method match
-            if let Some(glob_signal_pattern) = &self.signal {
-                // glob has signal defined
-                if let Some(ri_signal) = shv_ri.signal() {
-                    // RI has signal defined
-                    return glob_signal_pattern.matches(ri_signal);
-                } else {
-                    return false;
-                }
+        if !self.path.matches(shv_ri.path()) {
+            return false;
+        }
+        if !self.method.matches(shv_ri.method()) {
+            return false;
+        }
+        // path and method match
+        if let Some(glob_signal_pattern) = &self.signal {
+            // glob has signal defined
+            if let Some(ri_signal) = shv_ri.signal() {
```

Comment:
I THINK this nested if can be changed to a match, where you match both the `Some`s at the same time, right? But I don't know if it makes it simpler. Something along the lines of
```rust
match (&self.signal, shv_ri.signal()) {
  Some(glob_signal_patten, ri_signal) => glob_signal_pattern.matches(ri_signal),
  Some(glob_signal_pattern), None => false
  _ => true
}
```
I think it looks nice, but it's just my opinion
}

Reply:
Yeah, you're right 👍